lib/ansible/modules/storage/netapp/na_ontap_aggregate.py

```diff
@@ -0,0 +1,358 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: na_ontap_aggregate
+
+short_description: Manage NetApp ONTAP aggregates.
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: Sumit Kumar (sumit4@netapp.com), Suhas Bangalore Shekar (bsuhas@netapp.com)
+
+description:
+- Create or destroy aggregates on NetApp cDOT.
+
+options:
+
+  state:
+    description:
+    - Whether the specified aggregate should exist or not.
+    choices: ['present', 'absent']
+    default: 'present'
+
+  service_state:
+    description:
+    - Whether the specified aggregate should be enabled or disabled. Creates aggregate if doesnt exist.
+    required: false
+    choices: ['online', 'offline']
+
+  name:
+    required: true
+    description:
+    - The name of the aggregate to manage.
+
+  rename:
+    required: false
+    description:
+    - The name of the aggregate that replaces the current name.
+
+  disk_count:
+    description:
+    - Number of disks to place into the aggregate, including parity disks.
+    - The disks in this newly-created aggregate come from the spare disk pool.
+    - The smallest disks in this pool join the aggregate first, unless the C(disk-size) argument is provided.
+    - Either C(disk-count) or C(disks) must be supplied. Range [0..2^31-1].
+    - Required when C(state=present).
+
+  unmount_volumes:
+    required: false
+    description:
+    - If set to "TRUE", this option specifies that all of the volumes hosted by the given aggregate are to be unmounted
```

Comment:
Passing these as a list will cause them to render as multiple paragraphs. Use one of the YAML folded string formats to break lines for source readability/Python line length sanity checks.